app.py

```diff
@@ -1,5 +1,9 @@
+# -*- coding: utf-8 -*-
+
 from flask import Flask, render_template, request
+import requests
 from firebase import firebase
+import itertools
```

Comment:
In general, the order of imports should be standard library imports, then third-party imports. So this should look like:

```python
import itertools

from flask import Flask, render_template, request
from firebase import firebase
import requests
```

I recommend using https://github.com/PyCQA/flake8-import-order to enforce this.